### ISSUE TITLE
onLimitReached never called

### ISSUE THREAD

**Issue by aminosman** [nfriedly/express-rate-limit#149]
...  
     onLimitReached: (req) => {
            console.log(req.body)
            console.log(req.rateLimit)
     }
     ...

I am providing this for `onLimitReached` but this is never called. Change this to `handler` and it logs on each call over the rate limit. Any thoughts?

**Comment by aminosman** [comment]
Issue was my max was a fraction as I was calculating it, must be rounded for the `current === max + 1`  logic

**Comment by nfriedly** [comment]
Aah, good find. We should put in something that checks that `max` is a non-negative integer and throws if it isn't so that nobody else has that problem. Do you want to send a PR?

**Comment by aminosman** [comment]
We are in the middle of dealing with a spam attack, as soon as that is taken care of I will!

**Comment by martinburger** [comment]
Might be helpful for others who stumbled upon that issue:

I read the value of `options.max` from `process.env.RATE_LIMIT_MAX_REQUESTS`:

```javascript
const RATE_LIMIT_MAX_REQUESTS = process.env.RATE_LIMIT_MAX_REQUESTS;
```

It took me quite some time to figure out why `onLimitReached` was never called -- although the rate limiter blocked requests as expected as soon as that max limit had been reached.

As it turned out, that strange behavior was caused by`process.env.RATE_LIMIT_MAX_REQUESTS` being of type `string`.

Thus, this fixes the issue:

```javascript
const RATE_LIMIT_MAX_REQUESTS = Number(process.env.RATE_LIMIT_MAX_REQUESTS);
```

Bottom line: `onLimitReached` requires `options.max` to be of type `number`.

**Comment by Michael55555** [comment]
I've lost a significant amount of time today when I stumbled over this error. I'd recommend creating a warning if someone doesn't set the max option to a number.
